### PR TITLE
Add unit tests for replication addon

### PR DESCRIPTION
**Pull request by ELENAGER**
Signed-off-by: Elena Gershkovich <elenage@il.ibm.com>

### CODE REVIEW

File: controllers/tests/controller_server/addons_server_test.py [hunk 1]
```diff
@@ -41,34 +41,183 @@ def setUp(self):
...
+
+        self.assertEqual(grpc_status, self.context.code)
+        self.mediator.get_replication.assert_called_once_with(replication_request)
+        self.mediator.create_replication.assert_not_called()
+
+    def _test_disable_replication_succeeds(self, replication_type):
+        replication_request = self._prepare_request_params(replication_type)
+        self._prepare_replication_mocks(replication_type=replication_type)
+
+        self.servicer.DisableVolumeReplication(self.request, self.context)
```
Comment (ArbelNathan) [inline controllers/tests/controller_server/addons_server_test.py:105]: you are in class `TestControllerServicerEnableVolumeReplication` not `Disable`
please split all the tests into the correct classes

Comment (ELENAGER) [reply controllers/tests/controller_server/addons_server_test.py:158]: Added separate classes for Enable, disable, promote, demote and resync replication

File: controllers/tests/controller_server/addons_server_test.py [hunk 2]
```diff
@@ -41,34 +34,90 @@ def setUp(self):
...
 
         self.servicer.EnableVolumeReplication(self.request, self.context)
 
         self.assertEqual(grpc.StatusCode.OK, self.context.code)
-        replication_request = ReplicationRequest(OBJECT_INTERNAL_ID, OTHER_OBJECT_INTERNAL_ID, SYSTEM_ID, COPY_TYPE,
-                                                 REPLICATION_TYPE_MIRROR)
         self.mediator.get_replication.assert_called_once_with(replication_request)
         self.mediator.create_replication.assert_called_once_with(replication_request)
 
+    def _test_enable_replication_no_volume_fails(self):
```
Comment (matancarmeli7) [inline controllers/tests/controller_server/addons_server_test.py:87]: I am not sure but by [this](https://github.com/IBM/ibm-block-csi-driver/blob/6213d0a4386b22cdfbee3ebf8113d141be7691b1/controllers/servers/csi/addons_server.py#L32) line, I think it raises an error, if this true, can you add an assert for exception `ObjectNotFoundError`?

Comment (matancarmeli7) [reply controllers/tests/controller_server/addons_server_test.py:87]: my bad

**Review by matancarmeli7 — DISMISSED** [review]
LGTM

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Review by matancarmeli7 — APPROVED** [review]

**Pull request merged**